src/remote_adapter/remote_adapter_server/plugins/transport/tests/plugins/message_broker_test.cc

```diff
@@ -0,0 +1,299 @@
+      parameter_type(kTestAddress, param_types::STRING),
+      parameter_type(std::to_string(kMsgTestPort), param_types::INT)};
+
+  auto recive_handle = client.async_call(constants::receive, parameters);
+
+  recive_handle.wait();
+
+  return recive_handle.get().as<response_type>();
+}
+
+void MessageBroker_Test::Accept(array_signal &signals) {
+
+  auto const address = net::ip::make_address(kTestAddress);
+  auto const port = kMsgTestPort;
+
+  net::io_context ioc{1};
+  tcp::acceptor acceptor{ioc, {address, port}};
+  tcp::socket socket{ioc};
+
+  signals[0].set_value();
+
+  acceptor.accept(socket);
+
+  Session(socket, signals);
+}
+
+void MessageBroker_Test::Session(tcp::socket &socket, array_signal &signals) {
+
+  websocket::stream<tcp::socket> ws{std::move(socket)};
+  // Accept the websocket handshake
+  ws.accept();
+
+  signals[1].set_value();
+
+  beast::flat_buffer buffer;
+  ws.read(buffer);
+  // Echo the message back
+  ws.text(ws.got_text());
+  ws.write(buffer.data());
+
+  signals[2].set_value();
+
+  Wait();
+}
+
+void MessageBroker_Test::Wait() {
+  std::unique_lock<std::mutex> lck(mtx);
+  cv.wait(lck);
+}
+
+void MessageBroker_Test::Notify() {
+  std::unique_lock<std::mutex> lck(mtx);
+  cv.notify_all();
+}
+
+TEST_F(MessageBroker_Test, CreatePlugin_Expect_TRUE) {
+  auto message_broker = create_plugin();
+  EXPECT_TRUE(message_broker);
+}
+
+TEST_F(MessageBroker_Test, DeletePlugin_Expect_TRUE) {
+  auto message_broker = create_plugin();
+  message_broker.reset();
+  EXPECT_TRUE(!message_broker);
+}
+
+TEST_F(MessageBroker_Test, Check_TypeId_Expect_EQ) {
+  using namespace msg_wrappers;
+  using namespace remote_adapter;
+  UtilsPlugin *message_broker_1 = new MessageBroker<>();
+  auto message_broker_2 = create_plugin();
+  EXPECT_EQ(typeid(message_broker_1), typeid(message_broker_2.get()));
+  delete message_broker_1;
+}
+
+TEST_F(MessageBroker_Test, OpenConnection_Expect_SUCCESS) {
+
+  auto message_broker = create_plugin();
+  EXPECT_TRUE(message_broker);
+
+  rpc::server server(kTestAddress, kRpcTestPort);
+
+  message_broker->Bind(server);
+
+  server.async_run();
+
+  rpc::client client(kTestAddress, kRpcTestPort);
+
+  auto response = OpenConnection(client);
+
+  EXPECT_EQ(constants::error_codes::SUCCESS, response.second);
+}
+
+TEST_F(MessageBroker_Test, CloseConnection_Expect_NO_CONNECTION) {
+
+  auto message_broker = create_plugin();
+  EXPECT_TRUE(message_broker);
+
+  rpc::server server(kTestAddress, kRpcTestPort);
+
+  message_broker->Bind(server);
+
+  server.async_run();
+
+  rpc::client client(kTestAddress, kRpcTestPort);
+
+  std::vector<parameter_type> parameters = {
+      parameter_type(kTestAddress, param_types::STRING),
+      parameter_type(std::to_string(kMsgTestPort), param_types::INT)};
+
+  auto close_handle = client.async_call(constants::close_handle, parameters);
+  close_handle.wait();
+
+  try {
+    auto response = close_handle.get().as<response_type>();
```

Comment:
The unit test expects exception in this line
https://github.com/smartdevicelink/sdl_atf/blob/760962eab530e7eeff84730cc86e4e81a802b155/src/remote_adapter/remote_adapter_server/plugins/transport/tests/plugins/message_broker_test.cc#L211-L213
the test should be success only if we catch the exception. In other case when exception will not be raised in the below code test should fail

Reply:
To clarify, It looks like the test will pass if an exception is not thrown, since it will just reach the end of the test after this call. I would suggest adding the following call within the try block to explicitly fail:
```
...
    auto response = close_handle.get().as<response_type>();
    ADD_FAILURE();
...
```

Reply:
@jacobkeeler 
Thanks,fixed in:https://github.com/smartdevicelink/sdl_atf/pull/180/commits/d87da584ae8be32c0d32aea0b9659908cbbb0fea